packages/manager/apps/telecom/src/app/telecom/telephony/alias/portability/portabilities/portabilities.controller.js

```diff
@@ -164,4 +63,45 @@ export default class TelecomTelephonyAliasPortabilitiesCtrl {
       PORTABILITY_STATUS.formReceived,
     ].includes(this.numbers[index].lastStepDone.name);
   }
+
+  groupPortabilityByNumbers = function groupPortabilityByNumbers(
+    portabilities,
+  ) {
+    return portabilities.flatMap((portability) =>
+      portability.numbersList.map((number) => ({
+        number,
+        portability,
+        lastStepDone: portability.steps
+          .slice()
+          .reverse()
+          .find((step) => step.status === this.PORTABILITY_STEPS_STATUS.done),
+      })),
+    );
+  };
+
+  hasChangeRequired(porta, step) {
+    if (step) {
+      return (
+        step.status === this.PORTABILITY_STEPS_STATUS.error &&
+        porta.portability.changeRequired?.length > 0
+      );
+    }
+    return porta.portability.error && !porta.lastStepDone;
+  }
+
+  selectGuides() {
+    let portaFr = 0;
+    let portaBe = 0;
+    this.numbers.forEach((porta) => {
+      if (porta.portability.portabilityCountry === PORTABILITY_COUNTRY.FR) {
+        portaFr += 1;
+      } else {
+        portaBe += 1;
+      }
+    });
```

Comment:
you could use a reduce returning an object here  or a filter and calculate according to array length

Reply:
I changed by:

```
selectGuides() {
    const portaFr = this.numbers.filter(
      (porta) =>
        porta.portability.portabilityCountry === PORTABILITY_COUNTRY.FR,
    );

    return portaFr.length >= this.numbers / 2
      ? PORTABILITY_STEPS_GUIDE.FR
      : PORTABILITY_STEPS_GUIDE.BE;
  } 
  ```